settings.go

```diff
@@ -33,172 +33,239 @@ type SettingsService struct {
-	SlackAppSecret                            string            `json:"slack_app_secret"`
-	SlackAppVerificationToken                 string            `json:"slack_app_verification_token"`
-	SnowplowCollectorHostname                 string            `json:"snowplow_collector_hostname"`
-	SnowplowCookieDomain                      string            `json:"snowplow_cookie_domain"`
-	SnowplowEnabled                           bool              `json:"snowplow_enabled"`
-	SnowplowSiteID                            string            `json:"snowplow_site_id"`
-	TerminalMaxSessionTime                    int               `json:"terminal_max_session_time"`
-	Terms                                     string            `json:"terms"`
-	ThrottleAuthenticatedAPIEnabled           bool              `json:"throttle_authenticated_api_enabled"`
-	ThrottleAuthenticatedAPIPeriodInSeconds   int               `json:"throttle_authenticated_api_period_in_seconds"`
-	ThrottleAuthenticatedAPIRequestsPerPeriod int               `json:"throttle_authenticated_api_requests_per_period"`
-	ThrottleAuthenticatedWebEnabled           bool              `json:"throttle_authenticated_web_enabled"`
-	ThrottleAuthenticatedWebPeriodInSeconds   int               `json:"throttle_authenticated_web_period_in_seconds"`
-	ThrottleAuthenticatedWebRequestsPerPeriod int               `json:"throttle_authenticated_web_requests_per_period"`
-	ThrottleUnauthenticatedEnabled            bool              `json:"throttle_unauthenticated_enabled"`
-	ThrottleUnauthenticatedPeriodInSeconds    int               `json:"throttle_unauthenticated_period_in_seconds"`
-	ThrottleUnauthenticatedRequestsPerPeriod  int               `json:"throttle_unauthenticated_requests_per_period"`
-	TimeTrackingLimitToHours                  bool              `json:"time_tracking_limit_to_hours"`
-	TwoFactorGracePeriod                      int               `json:"two_factor_grace_period"`
-	UniqueIPsLimitEnabled                     bool              `json:"unique_ips_limit_enabled"`
-	UniqueIPsLimitPerUser                     int               `json:"unique_ips_limit_per_user"`
-	UniqueIPsLimitTimeWindow                  int               `json:"unique_ips_limit_time_window"`
-	UsagePingEnabled                          bool              `json:"usage_ping_enabled"`
-	UserDefaultExternal                       bool              `json:"user_default_external"`
-	UserDefaultInternalRegex                  string            `json:"user_default_internal_regex"`
-	UserOauthApplications                     bool              `json:"user_oauth_applications"`
-	UserShowAddSSHKeyMessage                  bool              `json:"user_show_add_ssh_key_message"`
-	VersionCheckEnabled                       bool              `json:"version_check_enabled"`
-	WebIDEClientsidePreviewEnabled            bool              `json:"web_ide_clientside_preview_enabled"`
+	ID                                                int               `json:"id"`
```

Comment:
I started to just take over the fields in the order they are in the GitLab doc, however I think that @svanharmelen usually wants them in alphabetical order (which I generally would prefer, too) ... so maybe we need to change that here :)

Reply:
That should be an easy fix. I need to make another pass at it today as I noticed a lot of missing fields when looking at a real reply from gitlab. Currently thinking that I might need to dig into the ruby code to get real answers 😅

Reply:
To save you the Ruby code search:

* [parameters for setting the settings](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/settings.rb#L33)
* [Entity returned when getting the settings](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/entities/application_setting.rb)
* [The visible attributes references in the code above](https://gitlab.com/gitlab-org/gitlab/-/blob/master/app/helpers/application_settings_helper.rb#L192)

Reply:
Yeah... Keeping the order consistent is an ongoing battle... Not too long ago in the docs a couple of locations started to use alphabetical ordering, but in the docs it's not really consistent.

Until now I generally try to follow to order used in the docs so that it's easy to lookup something (should be in the same position). But thinking about making a PR that just makes all structs (so both option and response structs) alphabetical to just have a single consistent approach in this package 🤷🏻 

So for now this order looks fine 👍🏻

Reply:
I know I'm not on the thread yet, but I'd personally suggest we don't use un-documented fields for exactly the reason you suggest. GitLab even has some APIs that are explicitly documented as internal, so they're not even internal and stable, they're just not documented 😕 

I'd say if we want to use them, we create an issue for GitLab and look to document them, which would allow their PM/engineers to chime in on whether they're stable.

Reply:
I think I agree with that...